### PR TITLE
For #26286 new web controls UI tests

### PR DESCRIPTION
**Pull request by AndiAJ**
For #26286 new web controls UI tests.

✅ All 10 UI tests successfully passed 100x on Firebase.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26286

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 1]
```diff
@@ -651,6 +656,233 @@ class BrowserRobot {
...
+
+    fun verifySelectedDate() {
+        for (i in 1..RETRY_COUNT) {
+            try {
+                assertTrue(
+                    mDevice.findObject(
+                        UiSelector()
+                            .text("Selected date is: $currentDate"),
+                    ).waitForExists(waitingTime),
+                )
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:746]: you forgot to add "break" here

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 2]
```diff
@@ -651,6 +656,233 @@ class BrowserRobot {
...
+            mDevice.waitForIdle(waitingTime)
+        } else {
+            clickPageObject(webPageItemWithResourceId("dropDown"))
+            mDevice.waitForIdle(waitingTime)
+        }
+    }
+
+    fun clickFormViewButton(button: String) {
+        val clockAndCalendarButton = mDevice.findObject(UiSelector().textContains(button))
+        clockAndCalendarButton.click()
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:684]: nit: mDevice.findObject(UiSelector().textContains(button)).click() without defining a val

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 3]
```diff
@@ -651,6 +656,233 @@ class BrowserRobot {
...
+            mDevice.waitForIdle(waitingTime)
+        } else if (clockForm) {
+            clickPageObject(webPageItemWithResourceId("clock"))
+            mDevice.waitForIdle(waitingTime)
+        } else if (colorPicker) {
+            clickPageObject(webPageItemWithResourceId("colorPicker"))
+            mDevice.waitForIdle(waitingTime)
+        } else {
+            clickPageObject(webPageItemWithResourceId("dropDown"))
+            mDevice.waitForIdle(waitingTime)
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:678]: these multiple if/else statements can be replaced with when expressions

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 4]
```diff
@@ -651,6 +656,233 @@ class BrowserRobot {
...
+        mDevice.findObject(UiSelector().resourceId("android:id/month_view")).waitForExists(waitingTime)
+
+        val monthViewerCurrentDay =
+            mDevice.findObject(
+                UiSelector()
+                    .textContains("$currentDay")
+                    .descriptionContains("$currentDay $currentMonth $currentYear"),
+            )
+
+        monthViewerCurrentDay.click()
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:697]: same as above, maybe not be worth defining the var if it's not re-usable, it just adds more code.

Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt]: Done !

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt (hunk 3 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt]: Done !

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt]: Done !

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:738]: Done !

**Comment by AndiAJ** [comment]
Thanks @sv-ohorvath , made the suggested changes re-ran all the tests 10x and work properly.

**Review by sv-ohorvath — APPROVED** [review]

**Review by AaronMT — APPROVED** [review]

**Pull request merged**